_posts/2016-08-28-katello-puppet4-integration.md

```diff
@@ -0,0 +1,122 @@
+tags:
+- katello
+- puppet 4
+excerpt: |
+    Puppet 4 integration is coming to Katello and we wanted to share with you
+    how it works!
+---
+
+### Puppet 4 Integration with Katello
+
+Recently Foreman has grown to work with Puppet 4. Katello is following suite and will soon be ready to use Puppet 4. In this post we will explain how to create new Katello environments Puppet 4 ready, and upgrade exisiting systems.
+
+There are some important things to know about environments with Katello and Puppet 4:
+
+Puppet 4 uses a Java Virtual Machine and requires more memory than Puppet 3 had before. This means that systems acting as Katello servers with Puppet 4 need a larger amount of memory. At a minimum we are requiring 8 GB of memory, and reccomending 16GB (<-- need a fact check on this). 
+
+Puppet 4 masters can still provision computers with Puppet 3 with [some configuration](https://docs.puppet.com/puppetserver/latest/compatibility_with_puppet_agent.html). This can help with systems that are migrating over to Puppet 4. You can start with upgrading your masters to Puppet 4 and then gradually upgrading you agents.
+
+You can use this matrix as a quick reference of what can work with what:
+
+| | Puppet 3 client | Puppet 4 client
+| :--- | :---: | :---: 
+| **Puppet 4 server** | Compatible | Compatible
+| **Puppet 3 server** | Compatible | Not compatible
+
+### How to install Katello with Puppet 4
+
+#### Production Environments
+If you wanted to create a new Katello server configured with Puppet 4 the easiest way to do this would be to spin up a Virtual Machine with [Forklift](https://github.com/Katello/forklift).
+
+To do this you can follow these instructions:
+
+```bash
+git clone https://github.com/katello/forklift.git
+cd forklift
+vagrant up centos7-katello-p4
+```
+
+You can also create a custom box by adding this configuration to your boxes.yml file:
+
+```yaml
+example-katello-p4:
+  box: centos7
+  memory: 9000
+  ansible: 
+    playbook: 'playbooks/capsule-dev.yml'
+    options: --puppet-four
+    group: 'capsule'
+    server: 'beav-dev-p4'
+```
+
+This will setup a virtual environment with Katello and Puppet 4 enabled.
+
+Then if you wanted to create a capsule with Forklift you can follow these steps:
+
+* In your boxes.yml file you can write
+```yaml
+example-capsule-p4:
+  box: centos7
+  ansible:
+    playbook: 'playbooks/capsule-dev.yml'
+    group: 'capsule'
+    server: 'example-katello-p4'
+```
+
+* After run:
+
+```bash
+vagrant up example-capsule-p4
+```
+
+#### Devlopment environment
+
+### Details on how to upgrade both Katello servers, capsules, and hosts to use puppet 4
+
+We have a [great video](https://youtu.be/GFNPHypFhl4?t=2545) and [excellent documentation](http://www.katello.org/docs/nightly/upgrade/puppet.html) on how to upgrade from Puppet 3 to 4.
+
+To upgrade an exisisting Katello instance to Puppet 4 it is straight forward:
+
+Before starting backup or take snapshot the Katello server.
```

Comment:
Maybe reword as well to "Before starting, we recommend creating a backup or taking a snapshot...`